x-pack/filebeat/input/s3/input.go

```diff
@@ -156,13 +139,32 @@ func NewInput(cfg *common.Config, connector channel.Connector, context input.Con
 	}
 
 	closeChannel := make(chan struct{})
+
+	// Wrap input.Context's Done channel with a context.Context. This goroutine
+	// stops with the parent closes the Done channel.
+	// Create a context with a timeout that will abort the API call if it takes
+	// more than the default timeout 2 minute.
+	ctx := context.Background()
+	var cancelFn func()
+	ctx, cancelFn = context.WithTimeout(ctx, config.ContextTimeout)
+	defer cancelFn()
```

Comment:
I think you can remove these, as the timeout should be added when doing API calls

Reply:
Yep you are right, this has been moved to where the APIs are actually called. Thanks!